src/main/java/com/epam/reportportal/cucumber/Utils.java

```diff
@@ -290,23 +292,23 @@ private static Method retrieveMethod(Match match) throws NoSuchFieldException, I
 	@Nullable
 	private static TestCaseIdEntry getTestCaseId(TestCaseId testCaseId, Method method, List<Argument> arguments) {
 		if (testCaseId.parametrized()) {
-			List<String> values = new ArrayList<String>(arguments.size());
+			List<String> values = new ArrayList<>(arguments.size());
 			for (Argument argument : arguments) {
 				values.add(argument.getVal());
 			}
 			return TestCaseIdUtils.getParameterizedTestCaseId(method, values.toArray());
 		} else {
-			return new TestCaseIdEntry(testCaseId.value(), testCaseId.hashCode());
+			return new TestCaseIdEntry(testCaseId.value());
 		}
 	}
 
 	private static TestCaseIdEntry getTestCaseId(String codeRef, List<Argument> arguments) {
-		List<String> values = new ArrayList<String>(arguments.size());
-		for (Argument argument : arguments) {
-			values.add(argument.getVal());
-		}
-		return new TestCaseIdEntry(StringUtils.join(codeRef, values.toArray()),
-				Arrays.deepHashCode(new Object[] { codeRef, values.toArray() })
-		);
+		return ofNullable(arguments).filter(args -> !args.isEmpty())
+				.map(args -> new TestCaseIdEntry(codeRef + TRANSFORM_PARAMETERS.apply(args)))
```

Comment:
any potential problem with spec symbols here?

Reply:
I suppose no, cause all symbols will be sent AS-IS. If user didn't specify testCaseId explicitly representation on the UI-side is not important for him and TestCaseHash will be generated regardless of TestCaseId content